packages/request-state/src/runtime/clientState.ts

```diff
@@ -8,36 +8,68 @@ const revivers: Record<string, (value: any) => any> = {
 	WellKnownSymbol: (value) => Symbol[value as keyof typeof Symbol],
 };
 
-const loadState = (doc: Document): State | undefined => {
-	const element = doc.getElementById('it-astro-state');
+const loadState = (doc: Document): State[] => {
+	const elements = Array.from(doc.querySelectorAll('.it-astro-state'));
+	const scripts = elements
+		.map((element) => {
+			if (element?.textContent) {
+				const state = parse(element.textContent, revivers);
+				element.remove();
+				return state;
+			}
+		})
+		.filter((state): state is State => !!state);
+	return scripts;
+};
+
+let nextStates = loadState(document);
 
-	if (element?.textContent) {
-		const state = parse(element.textContent, revivers);
-		element.remove();
-		return state;
+const initialiseState = (): State => {
+	const STATE_WINDOW_NAMESPACE = '__@it-astro:request-state-data' as const;
+	if (!(STATE_WINDOW_NAMESPACE in window)) {
+		window[STATE_WINDOW_NAMESPACE] = new Map();
 	}
+	return window[STATE_WINDOW_NAMESPACE] as State;
 };
 
-let nextState = loadState(document);
-const state = new Map();
+export const state = initialiseState();
 
-const applyState = () => {
-	const event = new ServerStateLoaded(new Map(state), nextState ?? new Map());
-
-	if (document.dispatchEvent(event)) {
-		state.clear();
-		for (const [key, value] of event.serverState.entries()) {
-			state.set(key, value);
+const mergeState = (oldState: State, newState: State) => {
+	for (const [newKey, newValue] of newState.entries()) {
+		const isHit = oldState.has(newKey);
+		if (isHit) {
+			if (process.env.NODE_ENV === 'development') {
+				console.warn(
+					`@inox-tools/request-state: tried to insert duplicate key ${newKey} with value ${newValue} but already had value ${oldState.get(newKey)}. Ignoring.`
+				);
+			}
+			continue;
 		}
+		oldState.set(newKey, newValue);
 	}
+	return oldState;
 };
 
-applyState();
+const applyState =
+	(isViewTransition: boolean = false) =>
+	() => {
+		const startingState = isViewTransition ? new Map() : new Map(state); // if it's an uncancelled view transition, we want a completely new state (otherwise the new view will 'lose' to the old view in the case of a conflict). If it's not a view transition, we want to shallow clone the state object. mergeState will update the top-level keys of the state passed to it, but if the user calls preventDefault on the serverStateLoaded we're supposed to abort the update. So it won't do to have already updated the underlying global state map. And on the other hand we do a shallow, not deep, clone because the user can mutate their own stored values any time they want.
```

Comment:
Put the comment on multiple lines above, not as a long trailing comment on a code line.